### DISCUSSION TITLE
raman.x

### DISCUSSION THREAD

**Discussion by kayojo** [nguyen-group/QERaman#17] [Q&A] [answered]
![WhatsApp Image 2024-05-13 at 7 36 45 PM](https://github.com/nguyen-group/QERaman/assets/165672975/586749af-ba85-4b3d-a234-4be67a719eb9)
I am running the raman.x on my HPC for the example and I got this error  
At line 152 of file raman_read.f90 (unit = 101)
Fortran runtime error: Cannot open file '': No such file or directory

Error termination. Backtrace:
#0  0x7ff056eadd11 in ???
#1  0x7ff056eae859 in ???
#2  0x7ff056eaf53f in ???
#3  0x7ff0570ee9c6 in ???
#4  0x7ff0570eecbc in ???
#5  0x55d8e4e26928 in __raman_read_MOD_read_dvec
        at /work/pi_mmayes_umassd_edu/qe-7.3/QERaman/src/raman_read.f90:152
#6  0x55d8e4e1ebae in raman
        at /work/pi_mmayes_umassd_edu/qe-7.3/QERaman/src/raman.f90:51
#7  0x55d8e4e1e23e in main
        at /work/pi_mmayes_umassd_edu/qe-7.3/QERaman/src/raman.f90:4

and the graphene.dvec file is present in my working directory i don't why it giving this error pls any help on this will be appreciated. I have also attached the image of the output file
![WhatsApp Image 2024-05-13 at 5 27 29 PM](https://github.com/nguyen-group/QERaman/assets/165672975/2b3ad242-c9e1-46c3-b9ef-8790edda65ae)

**Comment by nguyen-group** [comment]
Hi,

Could you follow these steps?
1) Check whether both files "graphene.dvec" and "graphene.elph" are in your working directory or not.
2) If yes, please check that you put the correct link folded to these files in Raman.in or not.
3) If yes, you might need to open these files to check whether the matrix elements are written or not.

I hope that these steps can help.

Best,
Nguyen

**Reply by kayojo** [reply]
yes both files are in the working directory and the matrix element was written.
about linking the fold i just used the input as it was in the bands.in 
&BANDS
  prefix   = 'graphene'
  outdir   = './tmp'
  filband  = 'graphene.bands'
  lp       = .true.
  filp     = 'graphene.dvec'
 DO i need to add some command to my bash script to make it read or access the file

**Comment by kayojo** [comment]
I solved this I think because I am working in an HPC environment the code was not reading my input file name for raman so I had to make some changes to the src raman.readf90 code. I will summarize what I did maybe it might be useful for someone. changes I made 
  to enable the src raman_read.f90 Fortran code to accept the input file name as a command-line argument, I made the following changes:
Add a Subroutine to Get the Command-Line Argument:
Added a subroutine get_input_file to read the input file name from the command line.
Update raman_readin to Use the Provided Input File:
Updated raman_readin to use the file name obtained from the command line. then I recompiled it

**Reply by nguyen-group** [reply]
Dear Kayojo,

I really appreciate your efforts. 
Your information is extremely helpful.

Best,
Nguyen

**Comment by kayojo** [comment]
you welcome professor, I wanted to ask you a question on quantum expresso, I noticed the constrained magnetization for bands code in quantum expresso &QERaman is not tested or allowed to run. so how do I go about calculating the band structure of a compound with more than one transition metal in it and can be in a high spin low-spin configuration. usually we use tot-magnetization to fix the spin configuration or magnetism but since it won't run for bands do you have an ideal on how to fix the magnetization for such calculation in bands so that we can calculate for compound like MnFeCN at different spin configuration. I was thinking of using starting magnetization, but it usually converges to mostly one spin configuration like only high spin-low spin, you might not be able to try for other spin configurations.

Best Regards,
Kayode.

________________________________

**Reply by nguyen-group** [reply]
Dear Kayode,

You are correct. The constrained magnetization is disabled in QE for electron-phonon interaction. 
One way to run it is that you need to comment on the following lines 114-116 in the QERaman/src/bands.f90. Then, you can recompile with "make all". 
Please see: https://github.com/nguyen-group/QERaman/discussions/1
However, since it is not tested, you might get the bug. 

Best,
Nguyen

**Comment by kayojo** [comment, marked as answer]
Thanks, professors.



Best Regards,
Kayode.
________________________________
